chandra_aca/plot.py

```diff
@@ -0,0 +1,266 @@
+    # mag 6 as 40 and mag 11 as 3
+    # interp should leave it at the bounding value outside
+    # the range
+    return np.interp(mag, [6.0, 11.0], [40.0, 3.0])
+
+
+def _plot_catalog_items(ax, catalog):
+    """
+    Plot catalog items (guide, acq, bot, mon, fid) in yang and zang on the supplied
+    axes object in place.
+
+    :param ax: matplotlib axes
+    :param catalog: data structure containing starcheck-style columns/attributes
+                    catalog records.  This can be anything that will work with
+                    astropy.table.Table(catalog).  A list of dicts is the convention.
+    """
+    cat = Table(catalog)
+    gui = cat[(cat['type'] == 'GUI') | (cat['type'] == 'BOT')]
+    acq = cat[(cat['type'] == 'ACQ') | (cat['type'] == 'BOT')]
+    fid = cat[cat['type'] == 'FID']
+    mon = cat[cat['type'] == 'MON']
+    for row in cat:
+        ax.annotate("%s" % row['idx'],
+                    xy=(row['yang'] - 120, row['zang'] + 60),
+                    color='red',
+                    fontsize=12)
+    ax.scatter(gui['yang'], gui['zang'],
+               facecolors='none',
+               edgecolors='green',
+               s=100)
+    for acq_star in acq:
+        acq_box = plt.Rectangle(
+            (acq_star['yang'] - acq_star['halfw'],
+             acq_star['zang'] - acq_star['halfw']),
+            width=acq_star['halfw'] * 2,
+            height=acq_star['halfw'] * 2,
+            color='blue',
+            fill=False)
+        ax.add_patch(acq_box)
+    for mon_box in mon:
+        # starcheck convention was to plot monitor boxes at 2X halfw
+        box = plt.Rectangle(
+            (mon_box['yang'] - (mon_box['halfw'] * 2),
+             mon_box['zang'] - (mon_box['halfw'] * 2)),
+            width=mon_box['halfw'] * 4,
+            height=mon_box['halfw'] * 4,
+            color='orange',
+            fill=False)
+        ax.add_patch(box)
+    ax.scatter(fid['yang'], fid['zang'],
+               facecolors='none',
+               edgecolors='red',
+               linewidth=.5,
+               marker='o',
+               s=175)
+    ax.scatter(fid['yang'], fid['zang'],
+               facecolors='none',
+               edgecolors='red',
+               marker='+',
+               linewidth=.5,
+               s=175)
+
+
+def _plot_field_stars(ax, stars, attitude, red_mag_lim=None, bad_stars=None):
+    """
+    Plot plot field stars in yang and zang on the supplied
+    axes object in place.
+
+    :param ax: matplotlib axes
+    :param stars: astropy.table compatible set of records of agasc entries of stars
+    :param attitude: Quaternion-compatible attitude
+    :param red_mag_lim: faint limit
+    :param bad_stars: boolean mask of stars to be plotted in red
+    """
+    stars = Table(stars)
+    quat = Quaternion.Quat(attitude)
+
+    # Add star Y angle and Z angle in arcsec to the stars table
+    yagzags = (radec2yagzag(star['RA_PMCORR'], star['DEC_PMCORR'], quat)
```

Comment:
> Also I think it would be reasonable if the input stars has yang and zang columns 
> to just use those without asking any more questions (i.e. without requiring an attitude or checking it).

OK!  I was afraid that I'd find a way to shoot myself in the foot if we mixed fixed basically fixed content with that very dynamic content in the stars table,  but I suppose it would take some work to break it that way.